pallets/loans-ref/src/lib.rs

```diff
@@ -584,6 +653,41 @@ mod pallet {
 			Ok(())
 		}
 
+		/// Propose a change an active loan modification.
+		/// The change is not performed until you call [`Pallet::modify()`].
+		#[pallet::weight(10_000)]
+		#[transactional]
+		pub fn propose_modification(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			loan_id: T::LoanId,
+			mutation: Mutation<T::Rate>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			// TODO: check the correct permission for each LoanChange.
```

Comment:
Maybe we need another role? or it can be the borrower cc @offerijns

Reply:
Currently we had discussed that this would be the `LoanAdmin` role. I need to think about this once more though to double check it makes sense, but let's go with that for now

Reply:
From here: https://docs.google.com/spreadsheets/d/1RJ5RLobAdumXUK7k_ugxy2eDAwI5akvtuqUM2Tyn5ts/edit?pli=1#gid=0

It seems like some changes depends on `LoanAdmin` and others on `PoolAdmin`

Reply:
@mustermeiszer, which should check this permission, `pallet-loans` or `ChangeGuard` implementation?

Reply:
> It seems like some changes depend on LoanAdmin and others on PoolAdmin

I think here you're right @offerijns and any property of a loan should be associated to a `LoanAdmin`.

> @mustermeiszer, which should check this permission, pallet-loans or ChangeGuard implementation?

This question no longer applies if we choose to have only one role